APIs/openEO/Glossary.qmd

```diff
@@ -0,0 +1,78 @@
+---
+title: "Glossary"
+format:
+  html:
+    code-fold: true
+jupyter: python3
+---
+
+
+The following glossary provides an introduction to the key technical terms commonly used when working with the openEO API.
+
+## General terms
+
+- **EO**: Earth Observation
+- **API**: Application Programming Interface ([wikipedia](https://en.wikipedia.org/wiki/Application_programming_interface){target="_blank"}); a communication protocol between client and back-end
+- **client**: Software tool or environment that end-users directly interact with, e.g. R (RStudio), Python (Jupyter notebook), and JavaScript (web browser); R and Python are two major data science platforms; JavaScript is a major language for web development
+- **(cloud) back-end**: server; computer infrastructure (one or more physical computers or virtual machines) used for storing EO data and processing it
+- **big Earth observation cloud back-end**: server infrastructure where industry and researchers analyse large amounts of EO data
+
+## Processes
+
+A **process** is an operation that performs a specific task on a set of parameters and returns a result. An example is computing a statistical operation, such as mean or median, on selected EO data. A process is similar to a function or method in programming languages.
+
+A **pre-defined process** is a process provided by the *back-end*, often one of the [ones centrally defined by openEO](https://openeo.org/documentation/1.0/processes.html){target="_blank"}.
+
+A [**user-defined process**](https://open-eo.github.io/openeo-python-client/udp.html){target="_blank"} is a process defined by the *user*. It can directly be part of another process graph or be stored as custom process on a back-end. Internally it is a *process graph* with optional additional metadata.
+
+A **process graph** chains specific process calls from the set of pre-defined and user-defined processes together. A process graph itself is a (user-defined) process again. Similarly to scripts in the context of programming, process graphs organize and automate the execution of one or more processes that could alternatively be executed individually. In a process graph, processes need to be specific, i.e. concrete values for input parameters need to be specified. These arguments can again be process graphs, scalar values, arrays or objects.
+
+## EO data (Collections)
+
+In our domain, different terms are used to describe EO data(sets). Within openEO, a **granule** (sometimes also called *item* or *asset* in the specification) typically refers to a limited area and a single overpass leading to a very short observation period (seconds) or a temporal aggregation of such data (e.g. for 16-day MODIS composites). A **collection** is a sequence of granules sharing the same product specification. It typically corresponds to the series of products derived from data acquired by a sensor on board a satellite and having the same mode of operation.
```

Comment:
This is probably copied from openeo.org, which is a bit outdated by now:

- Maybe start by stating that openEO simply follows STAC terminology. 
- Refer to, or copy explanations about collections/items/assets from STAC. e.g. https://github.com/radiantearth/stac-spec/blob/master/overview.md#what-is-a-spatiotemporal-asset
- opensearch is basically dead/non-existent as a standard, let's not reference it anymore as it would just build confusion 
- don't mention 'granule' anymore: it seems to have gone a bit out of use. I would mention 'item', as that is the STAC equivalent, or 'product'.

Reply:
updated